lib/Doctrine/ODM/MongoDB/Mapping/PropertyAccessors/RawValuePropertyAccessor.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\ODM\MongoDB\Mapping\PropertyAccessors;
+
+use Doctrine\ODM\MongoDB\Proxy\InternalProxy;
+use LogicException;
+use ProxyManager\Proxy\GhostObjectInterface;
+use ReflectionProperty;
+
+use function ltrim;
+
+use const PHP_VERSION_ID;
+
+/**
+ * This is a PHP 8.4 and up only class and replaces {@see ObjectCastPropertyAccessor}.
+ *
+ * It works based on the raw values of a property, which for a case of property hooks
+ * is the backed value. If we kept using setValue/getValue, this would go through the hooks,
+ * which potentially change the data.
+ */
+class RawValuePropertyAccessor implements PropertyAccessor
```

Comment:
This is not annotated as an internal class.

```suggestion
 *  
 * @internal
 */
class RawValuePropertyAccessor implements PropertyAccessor
```